index.ts

```diff
@@ -149,8 +158,34 @@ function findConfigFile(compiler: typeof typescript, searchPath: string, configF
 function ensureTypeScriptInstance(loaderOptions: LoaderOptions, loader: any): { instance?: TSInstance, error?: WebpackError } {
 
     function log(...messages: string[]): void {
+        logToConsole(stdoutConsole, messages);
+    }
+
+    function logToConsole(logConsole:any, messages: string[]): void {
         if (!loaderOptions.silent) {
-            console.log.apply(console, messages);
+            console.log.apply(logConsole, messages);
+        }
+    }
+
+    function logInfo(...messages: string[]): void {
+        if (LogLevel[loaderOptions.logLevel.toUpperCase()] <= LogLevel.INFO) {
```

Comment:
Rather than doing `.toUpperCase()` repeatedly here could we initialise the value of `logLevel` to the uppercase value when it is first read in?  That way the code will read a little cleaner and there will be a (super minor) perf gain.